enhance troubleshooting steps for edot java (#6931)
elastic/docs-content: troubleshoot/ingest/opentelemetry/edot-sdks/java/index.md

## Agent troubleshooting

> Determine if the issue is related to the EDOT agent, other agents, extensions or connectivity issues by following these steps:

Reviewer: The numbered list renders as 1, 1, 1 because the nested bullets are indented two spaces, which breaks the ordered list into separate single-item lists. Since this is a sequential troubleshooting flow, I'd suggest switching to stepper instead.

Alternatively, you could make each step a third-level heading (`### Run the application without the agent`, etc.)

As for stepper, it should look something like this (please note that I didn't apply any other comments to this):

```
Determine if the issue is related to the EDOT agent, other agents, extensions, or connectivity issues by following these steps:

:::::{stepper}

::::{step} Run the application without the agent
Start the application with no agent (remove the `-javaagent:` argument) and restart the application.

- If the issue is still present without the agent, then the problem is likely related to the application or its environment.
- If the issue is resolved, then re-adding the agent should allow to confirm that the issue is related to the agent.
::::

::::{step} Remove other agents
If more than one agent is being used on the application (multiple `-javaagent:` JVM arguments), remove other agents and restart the application. Multiple instrumentation agents can conflict with each other and are a known cause of unpredictable behavior.

- If the issue is still present with only the EDOT agent, then it indicates the problem is likely related to the agent.
- If the issue is resolved, then it indicates the problem might be related to interactions between agents.
::::

::::{step} Remove instrumentation extensions
If [instrumentation extensions](https://opentelemetry.io/docs/zero-code/java/agent/extensions/) are being used with the `otel.javaagent.extensions` JVM system property or the `OTEL_JAVAAGENT_EXTENSIONS` environment variable, remove them and restart the application.

- If the issue is still present without the extensions, then it indicates the problem is likely related to the agent.
- If the issue is resolved, then it indicates the problem might be related to the extensions.
::::

::::{step} Check end-to-end connectivity
Check end-to-end connectivity without the agent by running one or more of the example apps in [elastic-otel-java](https://github.com/elastic/elastic-otel-java/blob/main/examples/troubleshooting/README.md). These use the OpenTelemetry SDK rather than the auto-instrumentation. They can confirm that the issue is specific to the Java agent or can otherwise identify that the issue is caused by something else.
::::

:::::
```

Author: Thanks @alexandra5000 ! This is much nicer than a simple list :-)

<img width="761" height="1141" alt="image" src="https://github.com/user-attachments/assets/56aa37f8-78d2-44fe-87fa-aba581d65440" />

> 1. Start the application with no agent (remove `-javaagent:` argument) and restart the application:

Reviewer: ```suggestion
1. Start the application with no agent (remove `-javaagent:` argument) and restart the application.
```

Change suggested for consistency with other steps.

  - If the issue is still present without the agent, then the problem is likely related to the application or its environment.

>   - If the issue is resolved, then re-adding the agent should allow to confirm that the issue is related to the agent.

Reviewer: ```suggestion
  - If the issue is resolved, then re-adding the agent should let you confirm that the issue is related to the agent.
```

> 1. If more than one agent is being used on the application (multiple `-javagent:` JVM arguments), remove other agents and restart the application. Multiple instrumentation agents can conflict with each other and is a know cause of unpredictable behavior

Reviewer 2: @SylvainJuge PTAL

> 1. If more than one agent is being used on the application (multiple `-javagent:` JVM arguments), remove other agents and restart the application. Multiple instrumentation agents can conflict with each other and is a know cause of unpredictable behavior

Author: Should be fixed now, thanks!

>   - If the issue is still present with only the EDOT agent, then it indicates the problem is likely related to the agent.

Reviewer: ```suggestion
  - If the issue is still present with only the EDOT agent, then it indicates the problem is likely related to the EDOT agent.
```

I know we're repeating "EDOT agent" this way, but I would keep it just for clarity. It's better to be repetitive than vague in this case.

Author: I agree, when writing this writing "agent" alone felt weird in the current context where this term is over-used.

  - If the issue is resolved, then it indicates the problem might be related to interactions between agents.
1. If [instrumentation extensions](https://opentelemetry.io/docs/zero-code/java/agent/extensions/) are being used with the `otel.javaagent.extensions` JVM system property or the `OTEL_JAVAAGENT_EXTENSIONS` environment variable, remove them and restart the application.

>   - If the issue is still present without the extensions, then it indicates the problem is likely related to the agent.

Reviewer: ```suggestion
  - If the issue is still present without the extensions, then the problem is likely related to the agent.
```

  - If the issue is resolved, then it indicates the problem might be related to the extensions

> 1. Check end-to-end connectivity without the agent by running one or more of the example apps in [elastic-otel-java](https://github.com/elastic/elastic-otel-java/blob/main/examples/troubleshooting/README.md). These use the OpenTelemetry SDK rather than the auto-instrumentation. They can confirm that the issue is specific to the Java agent or can otherwise identify that the issue is caused by something else.

Reviewer: ```suggestion
1. Check end-to-end connectivity without the agent by running one or more of the example apps in [elastic-otel-java](https://github.com/elastic/elastic-otel-java/blob/main/examples/troubleshooting/README.md). These use the OpenTelemetry SDK rather than the auto-instrumentation. They can help confirm whether the issue is specific to the Java agent or caused by something else.
```

This version is simpler and cleaner, and keeps the same meaning.
